models/user/user.go

```diff
@@ -420,6 +421,16 @@ func (u *User) GetDisplayName() string {
 	return u.Name
 }
 
+func (u *User) GetSearchNameHTML() template.HTML {
+	if setting.UI.DefaultShowFullName {
+		trimmed := strings.TrimSpace(u.FullName)
+		if len(trimmed) > 0 {
+			return template.HTML(fmt.Sprintf(`%s<span class="text search-fullname"> %s</span>`, u.Name, trimmed))
+		}
+	}
+	return template.HTML(u.Name)
```

Comment:
Hey, my name is `<script>alert(...)</script>`  (well, for fun)

Every string used in templates needs to be escaped.

Reply:
> Hey, my name is `<script>alert(...)</script>` (well, for fun)
> 
> Every string used in templates needs to be escaped.

Added a commit to fix this.